promql/functions.go

```diff
@@ -573,9 +573,23 @@ func funcAvgOverTime(vals []parser.Value, args parser.Expressions, enh *EvalNode
 		return vec, nil
 	}
 	return aggrOverTime(vals, enh, func(s Series) float64 {
-		var mean, count, c float64
+		var (
+			sum, mean, count, c float64
+			incrementalMean     bool
+		)
 		for _, f := range s.Floats {
 			count++
+			if !incrementalMean {
+				newSum, newC := kahanSumInc(f.F, sum, c)
+				if count == 1 || !math.IsInf(newSum, 0) {
```

Comment:
Why do we care about count=1 ?

Reply:
We would get division-by-zero problems below if the very first value to sum up is already ±Inf. I have added a comment to explain that. In case of the aggregators, the flow is different because we essentially deal with the first step during the initialization, which happens for all the different aggregations in the same way.